kube/pod_test.go

```diff
@@ -2047,28 +2066,191 @@ func TestPodCPUAHelmActive(t *testing.T) {
 	`, actual)
 }
 
-func TestGetSecurityContext(t *testing.T) {
+func TestGetSecurityContextCapList(t *testing.T) {
 	assert := assert.New(t)
 
 	role := podTemplateTestLoadRole(assert)
 	if role == nil {
 		return
 	}
 
-	sc := getSecurityContext(role)
-	if !assert.NotNil(sc) {
+	t.Run("Kube", func(t *testing.T) {
+		sc := getSecurityContext(role, false)
+		if !assert.NotNil(sc) {
+			return
+		}
+
+		actual, err := testhelpers.RoundtripKube(sc)
+		if !assert.NoError(err) {
+			return
+		}
+		testhelpers.IsYAMLEqualString(assert, `---
+			capabilities:
+				add:
+				-	"SOMETHING"
+		`, actual)
+	})
+
+	t.Run("Helm", func(t *testing.T) {
+		sc := getSecurityContext(role, true)
+		if !assert.NotNil(sc) {
+			return
+		}
+
+		t.Run("OverrideNone", func(t *testing.T) {
+			config := map[string]interface{}{
+				"Values.sizing.myrole.capabilities": []interface{}{},
+			}
+			actual, err := testhelpers.RoundtripNode(sc, config)
+			if !assert.NoError(err) {
+				return
+			}
+			testhelpers.IsYAMLEqualString(assert, `---
+				capabilities:
+					add:
+					-	"SOMETHING"
+			`, actual)
+		})
+
+		t.Run("OverrideALL", func(t *testing.T) {
+			config := map[string]interface{}{
+				"Values.sizing.myrole.capabilities": []interface{}{"ALL"},
+			}
+			actual, err := testhelpers.RoundtripNode(sc, config)
+			if !assert.NoError(err) {
+				return
+			}
+			testhelpers.IsYAMLEqualString(assert, `---
+				privileged: true
+			`, actual)
+		})
+
+		t.Run("OverrideSomething", func(t *testing.T) {
+			config := map[string]interface{}{
+				"Values.sizing.myrole.capabilities": []interface{}{"something"},
+			}
+			actual, err := testhelpers.RoundtripNode(sc, config)
+			if !assert.NoError(err) {
+				return
+			}
+			testhelpers.IsYAMLEqualString(assert, `---
+				capabilities:
+					add:
+					-	"SOMETHING"
+					-	"something"
```

Comment:
I wonder if it would make more sense to use `assert.Contains` instead (walking down the struct instead of comparing YAMLs)

Reply:
While walking the structure directly might be more efficient I like the readability of the current code.
It feels to be (very) near to go's example-based testing facilities.